### PR TITLE
Fix documentation links always pointing to master.

### LINKED ISSUE #7988: [Good first issue] Documentation links in older docs versions
# 🚀 Feature request

This is a documentation request in order to make it easier to find corresponding examples in the documentation. 

Good first issue if you want to get acquainted with the docs and how to build docs using Sphinx!

## Current issue

Here's the issue: currently, if one goes to an older documentation version to check the "examples" page, for example, [v2.6.0](https://huggingface.co/transformers/v2.6.0/examples.html), all links point towards the `master` branch. 

For example, the link towards `run_tf_glue.py` is the following: https://github.com/huggingface/transformers/blob/master/examples/run_tf_glue.py

As this points towards the `master` branch, it is prone to breaking as files can (and probably will) be moved around as versions come out. It is the case for this example, as the `run_tf_glue.py` script is not in `examples/` anymore, but in `examples/text-classification/`. 

I think we need a way to ensure that all links point toward their appropriate version, and the easiest would be to point to a given tag. Since we're looking at the version `v2.6.0`, it makes sense to point towards the tag v2.6.0: https://github.com/huggingface/transformers/blob/v2.6.0/examples/run_tf_glue.py

This way links get frozen in time and redirect to actual files corresponding to their description and behaviour as stated in the docs.

## Resolution

I believe the easiest change would be to use sphinx variables in order to do this. Probably either [rst_epilog](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_epilog) or [rst_prolog](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_prolog) could be useful here.

Some useful links: [rst_epilog](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_epilog), [rst_prolog](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_prolog)

### PR DESCRIPTION
**Pull request by sugeeth14**
# What does this PR do?

This PR fixes documentation issue. The issue is that hyperlinks pointing to code file in the docs always point to the master.
Reference issue : https://github.com/huggingface/transformers/issues/7988

I found [extlinks](https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html) more appropriate than [rst_epilog
](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_epilog) as mentioned in the original issue because hyperlink replacement is not possible with `rst_epilog` 
Stackoverflow issue reference : https://stackoverflow.com/questions/1227037/substitutions-inside-links-in-rest-sphinx

Although this fixes the issues as discussed I am unable to fix the  issue  in markup files which are https://huggingface.co/transformers/master/contributing.html and https://huggingface.co/transformers/master/notebooks.html
Please suggest if it is possible to do that as well. 

Fixes https://github.com/huggingface/transformers/issues/7988

@sgugger @LysandreJik

### CODE REVIEW

File: docs/source/conf.py
Comment (sgugger) [inline docs/source/conf.py:31]: I think what @LysandreJik had in mind here was to use the `release` variable so each time we generate the docs on a given commit, the links are prefixed for that version.

But at the same time, it would also mean the master docs would always point to the latest release which is also something we don't want. So I think this line should be here but commented out, so that we uncomment it on the commit that is used to generate the docs of a specific version (with the proper tag using the `release` variable). This way when we regularly build the master doc at each PR, the links point to master, but when we make a release, we can just uncomment to build the links to the version released.

Does that make sense to you?

**Review by sgugger — APPROVED** [review]
Thanks for tackling this! The solution you propose does not solve all the issues around properly linking the tagged versions of the repo but I think we can change it slightly to make it work!

This is not urgent as @LysandreJik is on vacation (until Jan 4th) and I would very much like his review on this before merging.

File: docs/source/conf.py
Comment (Raghava14) [reply docs/source/conf.py]: I have made changes and committed. Is this what you wanted ? 

https://github.com/huggingface/transformers/pull/9217/files#diff-008dcb3426febd767787b1521f1fe33086313b927ea37eaab86df5fa88a51698R31

Comment (sgugger) [reply docs/source/conf.py]: Yes, it's good! Thanks a lot! Let's wait for @LysandreJik to be back to merge this if you don't mind.

Comment (Raghava14) [reply docs/source/conf.py]: Yeah @sgugger I don't mind. We can wait.

File: docs/source/model_doc/marian.rst
Comment (sgugger) [inline docs/source/model_doc/marian.rst:64]: Those are all missing `__` at the end and can probably fit in one line now.

File: docs/source/conf.py
Comment (sgugger) [reply docs/source/conf.py]: In the meantime, there seems to be some styling problems so make sure to run `make style` on your branch, and one error in the doc building step (I've added a comment in marian.rst where I think it comes from).

Comment (Raghava14) [reply docs/source/conf.py]: Yeah was checking those,thought of fixing them in the meantime. Thanks!

Comment (Raghava14) [reply docs/source/conf.py]: Why does `make style ` fix a lot of files which I haven't changed at all also when I try to use `style_doc.py` I get an error as here https://app.circleci.com/pipelines/github/huggingface/transformers/17685/workflows/f1f56681-8976-402a-95bd-387bd066ccb7/jobs/141134 that 10 files shoud be restyled ?! Am I missing something here ?
image

Comment (Raghava14) [reply docs/source/conf.py]: If I just do black `docs/source/conf.py` also there are changes from single quote to double quote 
image

**Review by LysandreJik — APPROVED** [review]
Hi, thanks for your work on this! I fixed the style and added a missing backtick so that all the tests pass. 

Great work, thanks @sugeeth14!

**Comment by sugeeth14** [comment]
Thanks for merging @LysandreJik but I want to know why the errors occurred when I ran `make style ` as I see in https://github.com/huggingface/transformers/pull/9217#discussion_r548898170

**Pull request merged**
